### ISSUE TITLE
[bitnami/mongodb] Password in custom users ServiceBinding changing after upgrade

### ISSUE THREAD

**Issue by alexweidemann** [bitnami/charts#18347]
### Name and Version

bitnami/mongodb

### What architecture are you using?

amd64

### What steps will reproduce the bug?

I am deploying the MongoDB Helm Chart with the values file below via the Helm CLI. After redeploying the Chart the password values in the ServiceBinding secrets of the custom users are changed and are not equal to the value of `mongodb-passwords` in the auth secret anymore.

Steps to reproduce:
1. `helm install my-release oci://registry-1.docker.io/bitnamicharts/mongodb -f path/to/values.yaml`
2. Compare values of `mongodb-passwords` in secret `my-release-mongodb` and of `password` in secret `my-release-mongodb-svcbind-0` --> Passwords are equal
3. `helm upgrade my-release oci://registry-1.docker.io/bitnamicharts/mongodb -f path/to/values.yaml`
4. Compare values of `mongodb-passwords` in secret `my-release-mongodb` and of `password` in secret `my-release-mongodb-svcbind-0` --> Passwords are *not* equal anymore

### Are you using any custom parameters or values?

`values.yaml`
```yaml
architecture: standalone
persistence:
  enabled: true
  storageClass: "my-storage-class"
  size: "2Gi"
auth:
  usernames:
    - app
  databases:
    - app
readinessProbe:
  enabled: true
startupProbe:
  enabled: true
serviceBindings:
  enabled: true
```

### What is the expected behavior?

I expected the `password` value in the ServiceBinding secret (in this case `my-release-mongodb-svcbind-0`) to not change and to be still equal to the `mongodb-passwords`  value in the auth secret (in this case `my-release-mongodb`) after executing `helm upgrade`.

### What do you see instead?

After executing `helm upgrade` the secret in `my-release-mongodb-svcbind-0` changed and the application using the MongoDB was not longer be able to access it because the authentication failed.

**Comment by CeliaGMqrz** [comment]
Hi @alexweidemann,

Thanks for reporting this issue. 

Indeed, I have been able to reproduce the problem. The passwords do not match after the chart upgrade. It seems that the logic in the [secret.yaml](https://github.com/bitnami/charts/blob/main/bitnami/mongodb/templates/secrets.yaml#L49) does not work correctly.

I have created an internal task to resolve this. We'll leave this issue on hold until we have updates.
